integrations/access/accessrequest/app.go

```diff
@@ -145,8 +157,23 @@ func (a *App) run(ctx context.Context) error {
 		return trace.Wrap(err)
 	}
 
-	if err := a.initAccessMonitoringRulesCache(ctx); err != nil {
-		return trace.Wrap(err)
+	r, err := retryutils.NewLinear(retryutils.LinearConfig{
+		Step: time.Second,
+		Max:  watchInitTimeout,
+	})
+	if err != nil {
+		return trace.Wrap(err, "getting watch kinds")
+	}
+	r.For(ctx, func() error {
+		if len(acceptedWatchKinds) == 0 {
+			return errors.New("no acceptedWatchKinds returned")
+		}
+		return nil
+	})
+	if slices.Contains(acceptedWatchKinds, types.KindAccessMonitoringRule) {
+		if err := a.initAccessMonitoringRulesCache(ctx); err != nil {
+			return trace.Wrap(err, "initializing Access Monitoring Rule cache")
+		}
 	}
```

Comment:
This implementation looks a bit awkward tbh, I thought previous iteration that used channel made more sense.

@fspmarshall Do you mind reviewing this PR? What would be the preferred way to gracefully ignore failure in watch of one of resources?

Reply:
After `job.WaitReady` returns, we know the callback function was executed, and so the `acceptedWatchKinds` if filled with the correct watch kinds.
We can remove the retry utility method.